apps/multiplatform/common/src/commonMain/kotlin/chat/simplex/common/views/chatlist/ChatListView.kt

```diff
@@ -298,15 +313,120 @@ fun connectIfOpenedViaUri(rhId: Long?, uri: URI, chatModel: ChatModel) {
     chatModel.appOpenUrl.value = rhId to uri
   } else {
     withApi {
-      planAndConnect(chatModel, rhId, uri, incognito = null, close = null)
+      planAndConnect(rhId, uri, incognito = null, close = null)
     }
   }
 }
 
+@Composable
+private fun ChatListSearchBar(listState: LazyListState, searchText: MutableState<TextFieldValue>, searchShowingSimplexLink: MutableState<Boolean>, searchChatFilteredBySimplexLink: MutableState<String?>) {
+  Row(verticalAlignment = Alignment.CenterVertically) {
+    val focusRequester = remember { FocusRequester() }
+    var focused by remember { mutableStateOf(false) }
+    Icon(painterResource(MR.images.ic_search), null, Modifier.padding(horizontal = DEFAULT_PADDING_HALF), tint = MaterialTheme.colors.secondary)
+    SearchTextField(
+      Modifier.weight(1f).onFocusChanged { focused = it.hasFocus }.focusRequester(focusRequester),
+      placeholder = stringResource(MR.strings.search_or_paste_simplex_link),
+      alwaysVisible = true,
+      searchText = searchText,
+      enabled = !remember { searchShowingSimplexLink }.value,
+      trailingContent = null,
+    ) {
+      searchText.value = searchText.value.copy(it)
+    }
+    val hasText = remember { derivedStateOf { searchText.value.text.isNotEmpty() } }
+    if (hasText.value) {
+      val hideSearchOnBack: () -> Unit = { searchText.value = TextFieldValue() }
+      BackHandler(onBack = hideSearchOnBack)
+      KeyChangeEffect(chatModel.currentRemoteHost.value) {
+        hideSearchOnBack()
+      }
+    } else {
+      Row {
+        val padding = if (appPlatform.isDesktop) 0.dp else 7.dp
+        val clipboard = LocalClipboardManager.current
+        val clipboardHasText = remember(focused) { chatModel.clipboardHasText }.value
+        if (clipboardHasText) {
+          IconButton(
+            onClick = { searchText.value = searchText.value.copy(clipboard.getText()?.text ?: return@IconButton) },
```

Comment:
when pasting a link to a known contact or group, it's filtered in chat list for a moment, then search is cleared and filter is reset.

This happens with links to others' contacts (known via address) and groups (known via group link); link to own group correctly stays filtered.